<table align="center" width=100%>
    <tr>
        <td width="10%">
            <img src="in_class.png">
        </td>
        <td>
            <div align="center">
                <font color="#21618C" size=8px>
                    <b> Inclass-Lab  <br>(Day 2)
                    </b>
                </font>
            </div>
        </td>
    </tr>
</table>

## Table of Content

1. **[Parameter Estimation](#est)**
2. **[Hypothesis Tests with Z Statistic](#eg)**
3. **[Error](#error)**

**Import the required libraries**

In [1]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import warnings
warnings.filterwarnings(action='ignore')

In [2]:
from math import factorial
from scipy import stats
from scipy.stats import randint
from scipy.stats import skewnorm
import random

In [3]:
from statsmodels.stats import weightstats as stests

### Let's begin with some hands-on practice exercises

<a id = "est"> </a>
## 1. Parameter Estimation

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>1. The quality controller at the automobile company needs to know the average length of a steel rod produced in the company. He managed to collect the length (in cm) of 15 rods produced in the last week. Find the point estimate for the average length of the steel rod using sample data. </b>
                </font>
            </div>
        </td>
    </tr>
</table>

Given data:

        len_rod (cm) = [25.2, 26.3, 28, 21.9, 23.4, 24, 27.2, 23, 29.2, 28.7, 23.1, 23.5, 26.4, 22.8, 24.7]

In [6]:
len_rod  = [25.2, 26.3, 28, 21.9, 23.4, 24, 27.2, 23, 29.2, 28.7, 23.1, 23.5, 26.4, 22.8, 24.7]

In [7]:
s_mean=np.mean(len_rod)
s_mean

25.16

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>2. The production manager at the automobile company states that all the steel rods are produced with an average length of 26 cm. Use the data given in the previous question and calculate the sampling error for the mean.</b>
                </font>
            </div>
        </td>
    </tr>
</table>

Given data:

        len_rod (cm) = [25.2, 26.3, 28, 21.9, 23.4, 24, 27.2, 23, 29.2, 28.7, 23.1, 23.5, 26.4, 22.8, 24.7]

In [10]:
pop_mean=26
len_rod= [25.2, 26.3, 28, 21.9, 23.4, 24, 27.2, 23, 29.2, 28.7, 23.1, 23.5, 26.4, 22.8, 24.7]

In [12]:
s_mean=np.mean(len_rod)


In [13]:
np.abs(s_mean-pop_mean)

0.8399999999999999

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>3. The production manager at tea emporium wants to estimate the weight of a green tea bag. The previous study shows that the standard deviation of weight is 2.3 g. The manager collects 65 tea bags for the study. How much margin of error the manager should consider to estimate the weight with 99%  confidence. </b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [19]:
sd=2.3
n=65
alpha=0.01

In [20]:
z_alpha_by2=round(stats.norm.isf(q=0.01/2),2)
z_alpha_by2

2.58

In [22]:
x=(z_alpha_by2*sd)/np.sqrt(n)
round(x,2)

0.74

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>4. The bank manager has received several fraud complaints from past a few weeks. The report from accountant states that the standard deviation in frauds is 16. The manager is ready to consider the margin of error as 5. To estimate the average number of frauds with 90% confidence how many transactions should the manager consider?</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [28]:
sigma=16
me=5


z_alpha_by2=round(stats.norm.isf(q=0.1/2),4)
z_alpha_by2

1.6449

In [29]:
n=(z_alpha_by2**2)*(sigma**2)/(me**2)
round(n)

28

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>5. A paediatrician wants to check the amount of sugar in the 100g pack of baby food produced by KidsGrow company. The medical journal states that a standard deviation of sugar in 100g pack is 8g. The paediatrician collects 37 packets of baby food and found that the average sugar is 24g. Find the 90% confidence interval for the population mean.</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [30]:
n=37
sample_mean=24
std=8

In [31]:
interval=stats.norm.interval(0.90,loc=sample_mean,scale=std/np.sqrt(n))
interval

(22.684117098438822, 25.315882901561178)

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>6. The NY university has opened the post of Astrophysics professor. The total number of applications was 36. To check the authenticity of the applicants a sample of 10 applications was collected, out of which 3 applicants were found to be a fraud. Construct the 95% confidence interval for the population proportion.</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [32]:
sd=8
N=37
n=10
n_froud=3

In [35]:
samp_prop=n_froud/n
samp_prop

0.3

In [39]:
stats.norm.interval(0.95,loc=samp_prop,scale=np.sqrt((samp_prop*(1-samp_prop))/n))

(0.015974234910674623, 0.5840257650893254)

<a id = "eg"> </a>
## 2. Hypothesis Tests with Z Statistic

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>7. The temperature (in degree Celsius) at Manhattan is recorded for 45 winter days. A student from the Geography department wants to test the claim proposed by the weather department. Check whether the given data satisfies the condition of normality with 99% confidence?</b>
                </font>
            </div>
        </td>
    </tr>
</table>

Use the data given below:
        
    temp_winter =  [2.51, 2.6, 1.8, 2.61, 3.9, 1.7, 1.9, 3.1, 1.58, 2.85, 2.9, 1.25, 1.3, 2.4, 3.22, 2.56, 1.32, 1.2, 1.26, 
                    3.2, 2.7, 3.5, 2.8, 1.5, 3.8, 1.84, 3.31, 1.24, 1.75, 1.41, 1.57, 1.4, 1.49, 2.26, 2.5, 2.52, 3.42, 1.36, 
                    2.35, 2.47, 2.28, 2.22, 2.66, 3.43, 1.98]

In [4]:
temp_winter =  [2.51, 2.6, 1.8, 2.61, 3.9, 1.7, 1.9, 3.1, 1.58, 2.85, 2.9, 1.25, 1.3, 2.4, 3.22, 2.56, 1.32, 1.2, 1.26, 
                3.2, 2.7, 3.5, 2.8, 1.5, 3.8, 1.84, 3.31, 1.24, 1.75, 1.41, 1.57, 1.4, 1.49, 2.26, 2.5, 2.52, 3.42, 1.36, 
                2.35, 2.47, 2.28, 2.22, 2.66, 3.43, 1.98]

In [ ]:
h0: skewness= 0(normally distributed) ha: skewness!= 0(Not normally distributed)

In [6]:
pvalue=shapiro(temp_winter)
pvalue

ShapiroResult(statistic=0.9453739523887634, pvalue=0.03401985391974449)

the pvalue 0.34 is greater than 0.01,so we accept the null hypothisis the data is normally distributed .

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>8. Use the temperature data given in the previous question. The weather department claims that the average temperature in Manhattan is less than 2.8 degree celsius in winter. Test the department's claim using a critical value method with 99% confidence.</b>
                </font>
            </div>
        </td>
    </tr>
</table>

Use the data given below:
        
    temp_winter =  [2.51, 2.6, 1.8, 2.61, 3.9, 1.7, 1.9, 3.1, 1.58, 2.85, 2.9, 1.25, 1.3, 2.4, 3.22, 2.56, 1.32, 1.2, 1.26, 
                    3.2, 2.7, 3.5, 2.8, 1.5, 3.8, 1.84, 3.31, 1.24, 1.75, 1.41, 1.57, 1.4, 1.49, 2.26, 2.5, 2.52, 3.42, 1.36, 
                    2.35, 2.47, 2.28, 2.22, 2.66, 3.43, 1.98]

In [9]:
temp_winter =  [2.51, 2.6, 1.8, 2.61, 3.9, 1.7, 1.9, 3.1, 1.58, 2.85, 2.9, 1.25, 1.3, 2.4, 3.22, 2.56, 1.32, 1.2, 1.26, 
                3.2, 2.7, 3.5, 2.8, 1.5, 3.8, 1.84, 3.31, 1.24, 1.75, 1.41, 1.57, 1.4, 1.49, 2.26, 2.5, 2.52, 3.42, 1.36, 
                2.35, 2.47, 2.28, 2.22, 2.66, 3.43, 1.98]

In [5]:
from scipy.stats import shapiro

h0: skewness= 0(normally distributed)
ha: skewness!= 0(Not normally distributed)

In [12]:
pvalue=shapiro(temp_winter)
pvalue

ShapiroResult(statistic=0.9453739523887634, pvalue=0.03401985391974449)

In [ ]:
The p-value is 0.034 is greater than 0.01 so we accept the null hypothisis that the data is normally distrubetd.

h0: the average temperature is Manhatten >= 2.8 degree celsious in winter (mu>= 2.8)
ha: the average temperature is  Manhatten < 2.8 degree celsious in winter (mu<2.8)   

In [8]:
z_critical=round(stats.norm.isf(q=0.99),4)
z_critical

-2.3263

In [ ]:
if the z calculated is going to be less than the -2.3263,then we reject the null value hypothisis

In [13]:
x_bar=np.mean(temp_winter)

mu =2.8
n=len(temp_winter)
sd=np.std(temp_winter,ddof=1)

In [14]:
z_score=(x_bar - mu )/(sd/np.sqrt(n))
z_score

-4.439552007491998

so we reject the null hypothisis , accepet the alternet hypothesis.
the average temperature is  Manhatten <2.8 degree celsious in winter (mu<2.8)   


In [16]:
p_value=stats.norm.cdf(z_score)
p_value

4.507316464110269e-06

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>9. The bank manager claims that the complaints regarding fraud transactions are less than 12 per month with a standard deviation of 3 frauds. To test the claim the sample data for 3 years is collected. The average number of frauds in sample data is 10. Use the critical value method to test the claim with 95% confidence.</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [17]:
h0: mu>=12
ha: mu<12

In [20]:
z_critical=round(stats.norm.isf(q=0.95),2)
z_critical

-1.64

In [ ]:
acording h0 mu>=12 so we have to go let tail test 

In [43]:
mu=12
x_bar=10
std=3
no_n=36


In [45]:
z_critical=(x_bar - mu )/(std/np.sqrt(no_n))
z_critical

-4.0

In [ ]:
we reject the null hypothesis 

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>10. The internet provider claims that their router gives a speed of 25 Mbps with a standard deviation of 4.8 Mbps. The computer engineer wants to check whether the speed is different than what the provider claims. For this study, the engineer collects the speed of 40 routers and found that the average speed is 23.2 Mbps for those 40 routers. Test the claim of engineer with 90% confidence.</b>
                </font> 
            </div>
        </td>
    </tr>
</table>

In [23]:
h0: mu=25
h1: mu!=25

In [25]:
critical_value_z=round(stats.norm.isf(q=0.10/2),2)
critical_value_z

1.64

In [46]:

mu=25
std=4.8
xbar=23.2
no=40

In [48]:
critical_value_z=(xbar - mu )/(std/np.sqrt(no))
critical_value_z

-2.3717082451262854

here z score is less than -1.64 so we reject the null value hypothesis

the internet speed is not 25 mbps mu!=25

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>11. The health magazine in New York states that every person in NY drinks more than 1.8 L water every day with a standard deviation of 0.5 L. To test this claim the physician collects the data of 45 people and found that the average water intake for these people is 1.93 L. Use the p-value technique to test the claim with 95% confidence.</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [30]:
h0: mu<1.8
ha: mu>1.8

In [58]:
z_critical_value=round(stats.norm.isf(q=0.05),2)
z_critical_value

1.64

In [ ]:
we clearly say that right tail test

In [53]:
xbar=1.93
n=45
mu=1.8
sd=0.5

In [56]:
z_score_value=(xbar - mu )/(sd/np.sqrt(n))
z_score_value

1.7441330224498344

In [ ]:
reject the null hypothis.z_value<

In [57]:
1-stats.norm.cdf(z_score_value)

0.04056794851105716

In [ ]:
accept h1

<a id= "error"></a>
## 3. Error

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>12. The health magazine in New York states that every person in NY drinks more than 1.8 L water every day with a standard deviation of 0.5 L. To test this claim the physician collects the data of 45 people and found that the average water intake for these people is 1.93 L. Compute the type II error for the test; given that the type I error is 0.05.</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [60]:
h0: mu<1.8
ha: mu>1.8

In [65]:
from statsmodels.stats import power

In [66]:
n=45
xbar=1.93
mu=1.8
sd=0.5

In [69]:
power_of_the_test = power.zt_ind_solve_power(effect_size=(xbar -mu)/sd,
    nobs1=n,
    alpha=0.05,
    power=None,
    ratio=0,
    alternative='larger')

power_of_the_test

0.5395417812025957

In [70]:
type_2_error = 1-power_of_the_test
type_2_error


0.46045821879740434